src/Giveaway.js

```diff
@@ -441,155 +492,341 @@ class Giveaway extends EventEmitter {
+                            allowedMentions: this.allowedMentions
+                        });
+                    }
+                } else if (message?.length <= 2000) channel.send({ content: message, allowedMentions: this.allowedMentions });
                 resolve(winners);
             } else {
-                const embed = this.manager.generateNoValidParticipantsEndEmbed(this);
-                this.message.edit(this.messages.giveawayEnded, { embed }).catch(() => {});
+                const message = this.fillInString(noWinnerMessage?.content || noWinnerMessage);
+                const embed = this.fillInEmbed(noWinnerMessage?.embed);
+                if (message || embed) {
+                    channel.send({
+                        content: message,
+                        embeds: embed ? [embed] : null,
+                        allowedMentions: this.allowedMentions
+                    });
+                }
+
+                this.message.edit({
+                    content: this.fillInString(this.messages.giveawayEnded),
+                    embeds: [this.manager.generateNoValidParticipantsEndEmbed(this)],
+                    allowedMentions: this.allowedMentions
+                }).catch(() => {});
                 resolve([]);
             }
         });
     }
 
     /**
-     * Rerolls the giveaway
-     * @param {GiveawayRerollOptions} options
+     * Rerolls the giveaway.
+     * @param {GiveawayRerollOptions} [options] The reroll options.
      * @returns {Promise<Discord.GuildMember[]>}
      */
-    reroll(options) {
+    reroll(options = {}) {
         return new Promise(async (resolve, reject) => {
-            if (!this.ended) {
-                return reject('Giveaway with message ID ' + this.messageID + ' is not ended.');
-            }
-            if (!this.channel) {
-                return reject('Unable to get the channel of the giveaway with message ID ' + this.messageID + '.');
+            if (!this.ended) return reject('Giveaway with message Id ' + this.messageId + ' is not ended.');
+            this.message ??= await this.fetchMessage().catch(() => {});
+            if (!this.message) return reject('Unable to fetch message with Id ' + this.messageId + '.');
+            if (!options || typeof options !== 'object') return reject(`"options" is not an object (val=${options})`);
+            options = merge(GiveawayRerollOptions, options);
+            if (options.winnerCount && (!Number.isInteger(options.winnerCount) || options.winnerCount < 1)) {
+                return reject(`options.winnerCount is not a positive integer. (val=${options.winnerCount})`);
             }
-            await this.fetchMessage().catch(() => {});
-            if (!this.message) {
-                return reject('Unable to fetch message with ID ' + this.messageID + '.');
+            if (this.isDrop) {
+                return reject(`Cannot reroll a drop giveaway.`);
```

Comment:
is this just a preference of yours?
cause it code wise it would be possible right?
so why not allow it? it doesn't have to be used after all

Reply:
the goal of a drop is to be the fastest person to react. Why would we reroll it? it would be random, then it's not a drop anymore

Reply:
I guess that's true.
I just thought if somewhy a host considers a winner invalid.
But yeah if he rerolls it will be a random person. = not fair i guess
alright

Reply:
if I were a server admin, I would instead create a new giveaway